chat-components/src/services/BroadcastService.ts

```diff
@@ -45,7 +51,18 @@ export const BroadcastService = {
          * Omit copying methods to prevent 'DataCloneError' in older browsers when passing an object with functions
          * This exception occurs when an object can't be clone with the 'structured clone algorithm' (used by postMessage)
          */
-        pubChannel.postMessage(JSON.parse(JSON.stringify(message)));
+        try {
+            const messageCopy = JSON.parse(JSON.stringify(message));
```

Comment:
this line is prone to errors , should extract to a transform function? for safety